python-venv/cmstools/test/cfs_sessions_rc_test/__main__.py

```diff
@@ -0,0 +1,151 @@
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import argparse
+import sys
+from typing import NoReturn
+
+from cmstools.test.cfs_sessions_rc_test.helpers.cleanup import cleanup_and_restore
+from cmstools.test.cfs_sessions_rc_test.helpers.setup import get_cfs_config_name
+from cmstools.test.cfs_sessions_rc_test.helpers.setup import cfs_sessions_rc_test_setup
+from cmstools.test.cfs_sessions_rc_test.subtests import all_subtests
+from cmstools.lib.log import LOG_FILE_PATH
+
+from .log import logger
+from .defs import ScriptArgs, CFSRCException
+from .argument_parser import (add_mutually_exclusive_arguments, add_setup_arguments,
+                              add_subtests_arguments, add_cfs_session_arguments)
+
+
+def get_test_names(script_args: ScriptArgs) -> list[str]:
+    """Get the list of subtests to run or skip based on command line arguments."""
+    all_tests = list(all_subtests().keys())
+
+    if script_args.run_subtests:
+        return script_args.run_subtests
+
+    if script_args.skip_subtests:
+        return [name for name in all_tests if name not in script_args.skip_subtests]
+
+    return all_tests
+
+
+def _execute_subtests(test_names: list[str], script_args: ScriptArgs) -> None:
```

Comment:
Instead of passing around the test names and then having to map them back to the test classes here, would it make sense to have `get_test_names` instead be `get_tests`, and return the dictionary mapping instead of just the name list? It is already calling all_subtests to get that mapping, and would just need to return the subset of the mapping that corresponded to the tests actually being run, right?